packages/builder/src/pages/builder/app/[application]/design/_components/NewScreen/CreateScreenModal.svelte

```diff
@@ -117,105 +94,80 @@
-      let screenTemplate = template.create()
-      screenTemplate.autoTableId = template.resourceId
-      return screenTemplate
-    })
-    const createdScreens = await createScreens({ screens, screenAccessRole })
-    loadNewScreen(createdScreens)
-  }
+  const createBlankScreen = async ({ screenUrl }) => {
+    const screenTemplate = blankScreen(screenUrl)
+    const screen = await createScreen(screenTemplate)
+    await addNavigationLink(screenTemplate)
 
-  const confirmScreenBlank = async ({ screenUrl }) => {
-    blankScreenUrl = screenUrl
-    screenAccessRoleModal.show()
+    loadNewScreen(screen)
   }
 
-  // Submit request for a blank screen
-  const confirmBlankScreenCreation = async ({
-    screenUrl,
-    screenAccessRole,
-  }) => {
-    if (!pendingScreen) {
-      return
+  const createGridScreen = async () => {
+    let firstScreen = null
+
+    for (let datasource of selectedDatasources) {
+      const screenTemplate = gridScreen(
+        datasource,
+        await permissions.getResource(datasource.resourceId)
+      )
+
+      const screen = await createScreen(screenTemplate)
+      await addNavigationLink(screen)
+
+      firstScreen ??= screen
     }
-    pendingScreen.routing.route = screenUrl
-    const createdScreens = await createScreens({
-      screens: [pendingScreen],
-      screenAccessRole,
-    })
-    loadNewScreen(createdScreens)
-  }
 
-  const onConfirmFormType = () => {
-    screenAccessRoleModal.show()
+    loadNewScreen(firstScreen)
   }
 
-  const loadNewScreen = createdScreens => {
-    const lastScreen = createdScreens.slice(-1)[0]
+  const createGridDetailsScreen = async () => {
+    let firstScreen = null
 
-    // Go to new screen
-    if (lastScreen?.props?._children.length) {
-      // Focus on the main component for the streen type
-      const mainComponent = lastScreen?.props?._children?.[0]._id
-      $goto(`./${lastScreen._id}/${mainComponent}`)
-    } else {
-      $goto(`./${lastScreen._id}`)
+    for (let datasource of selectedDatasources) {
+      const screenTemplate = gridDetailsScreen(
+        datasource,
+        await permissions.getResource(datasource.resourceId)
+      )
+
+      const screen = await createScreen(screenTemplate)
+      await addNavigationLink(screen)
+
+      firstScreen ??= screen
     }
 
-    screenStore.select(lastScreen._id)
+    loadNewScreen(firstScreen)
   }
 
-  const confirmFormScreenCreation = async () => {
-    templates = formScreen(selectedDatasources, { actionType: formType })
-    screens = templates.map(template => {
-      let screenTemplate = template.create()
-      return screenTemplate
-    })
-    const createdScreens = await createScreens({ screens, screenAccessRole })
+  const createFormScreen = async formType => {
+    let firstScreen = null
+
+    for (let datasource of selectedDatasources) {
+      const screenTemplate = formScreen(
+        datasource,
+        formType,
+        await permissions.getResource(datasource.resourceId)
```

Comment:
Could we add a try/catch around some or all of these requests? There are a lot of them and from what I can see, at least of the `permissions.getResource` request, they aren't handled anywhere.